integrations/helm/chartsync/deps.go

```diff
@@ -14,6 +14,8 @@ func updateDependencies(chartDir, helmhome string) error {
 	// sanity check: does the chart directory exist
 	chartInfo, err := os.Stat(chartDir)
 	switch {
+	case os.IsNotExist(err):
```

Comment:
.. although, checking a possibly `nil` value here makes me nervous. We're relying on the implementation of `os.IsNotExist` being adequately defensive.

Reply:
I could apply the same safety net for this like we do during installation.

https://github.com/weaveworks/flux/blob/193dff6ee957e12f48428edd586472793d366bc3/integrations/helm/release/release.go#L128-#L130

Reply:
This should be more to your liking.

Reply:
I meant using `os.IsNotExist(err)` before checking whether `err != nil`

Reply:
Misread your initial comment...

We also do this in the `release.go` linked above and the example for `os.IsNotExist` (https://golang.org/pkg/os/#IsNotExist) uses the same approach, the latter giving me the impression that this is fine.